docs/static/breaking-changes.asciidoc

```diff
@@ -1,93 +1,256 @@
 [[breaking-changes]]
 == Breaking changes
 
-This section discusses the changes that you need to be aware of when migrating your application to Logstash {version}.
+This section discusses the changes that you need to be aware of when migrating your application to Logstash {version} from the previous major release of Logstash (2.x).
+
+[float]
+=== When to Upgrade
```

Comment:
Some of this information looks like it belongs in the upgrade doc, not the breaking changes topic. @acchen97 is in the process of scoping out what needs to change in the upgrade doc. The general upgrade stuff should be added to the new content that we add under this topic: https://www.elastic.co/guide/en/logstash/5.0/upgrading-logstash.html